clients/lsp-csharp.el

```diff
@@ -235,6 +241,99 @@ tarball or a zip file (based on a current platform) to TARGET-DIR."
   "Resolves path and arguments to use to start the server."
   (list (lsp-csharp--language-server-path) "-lsp"))
 
+(defun lsp-csharp--reset-test-buffer (present-buffer)
+  "Create new or reuse an existing test result output buffer.
+
+PRESENT-BUFFER will make the buffer be presented to the user."
+  (let ((existing-buffer (get-buffer lsp-csharp-test-run-buffer-name))
+        (project-root-dir (lsp--suggest-project-root)))
+    (if existing-buffer
+        (progn
+          (with-current-buffer existing-buffer
+            (setq buffer-read-only nil)
+            (erase-buffer)
+            (setq buffer-read-only t)
+            (setq default-directory project-root-dir))
+          existing-buffer)
+      (let ((buffer (get-buffer-create lsp-csharp-test-run-buffer-name)))
+        (with-current-buffer buffer
+          (setq default-directory project-root-dir)
+          (compilation-mode)
+          buffer))))
+
+  (if present-buffer
+      (display-buffer lsp-csharp-test-run-buffer-name)))
+
+(defun lsp-csharp--test-message (message)
+  "Emit a MESSAGE to lsp-csharp test run buffer."
+  (let ((existing-buffer (get-buffer lsp-csharp-test-run-buffer-name)))
+    (if existing-buffer
+        (with-current-buffer existing-buffer
```

Comment:
nit: typically, this is performed in save-excursion block. There is a similar code in `dap--print-to-output-buffer` to preserve the point depending on point position.